File: Python/matriz_quadrada.py
import numpy as np
import time

def multiplicar_matrizes(A, B):
    return A @ B

def main():
    tamanhos = [500, 1000, 1500, 2000, 2500, 3000]
    resultados = []

    with open('matriz_quadrada_python.txt', 'w') as f:
        for n in tamanhos:
            for j in range(1, 11):
                # Preenche as matrizes A e B com valores aleatórios
                A = np.random.rand(n, n)
                B = np.random.rand(n, n)

                start_time = time.perf_counter()
                C = multiplicar_matrizes(A, B)
                end_time = time.perf_counter()

                tempo = end_time - start_time
                mensagem = f"Tempo para matriz de tamanho {n} (teste {j}): {tempo:.4f}s"
                print(mensagem)
                f.write(mensagem + '\n')

if __name__ == "__main__":
    main()

File: NodeJS/matriz_quadrada.js
const math = require('mathjs');
const fs = require('fs');

function multiplicarMatrizes(A, B) {
    return math.multiply(A, B);
}

async function main() {
    const tamanhos = [500, 1000, 1500, 2000, 2500, 3000];

    for (const n of tamanhos) {
        for (let j = 1; j <= 10; j++) {
            const A = math.random([n, n]);
            const B = math.random([n, n]);

            const start = process.hrtime.bigint();
            multiplicarMatrizes(A, B);
            const end = process.hrtime.bigint();

            const tempo = Number(end - start) / 1e9;

            const mensagem = `Tempo para matriz de tamanho ${n} (teste ${j}): ${tempo.toFixed(4)}s`;
            console.log(mensagem);

            fs.appendFileSync('matriz_quadrada_node.txt', mensagem + '\n');
        }
    }
}

main();
